tests/all2all/alltoallconfig_test.go

```diff
@@ -0,0 +1,118 @@
+package all2all
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/relab/gorums"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var replicaCount = flag.Int("replicas", 20, "number of replicas to create all-to-all communication")
+
+func TestAllToAllConfiguration(t *testing.T) {
+	replicas, err := createReplicas(*replicaCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		for _, replica := range replicas {
+			replica.stopServer()
+		}
+	}()
+	nodeMap := make(map[string]uint32)
+	for _, replica := range replicas {
+		nodeMap[replica.address] = replica.id
+	}
+	for _, replica := range replicas {
+		replica.createConfiguration(nodeMap)
+	}
+}
+
+// createReplicas returns a slice of replicas.
+// The function waits for all serve goroutines to start and one additional
+// second to allow the servers to start before returning.
+func createReplicas(numReplicas int) ([]*replica, error) {
+	replicas := make([]*replica, 0)
+	errChan := make(chan error, numReplicas)
+	startedChan := make(chan struct{}, numReplicas)
+	startedCnt := 0
+	defer close(errChan)
+	for i := 1; i <= numReplicas; i++ {
```

Comment:
If we initialize replicas with a size, i.e. `replicas := make([]*replica, numReplicas)`, this for loop can be simplified to

```go
for i in range replicas {
  ...
  replicas[i] = &replica
  ...
}
```